screens/Map.js

```diff
@@ -69,43 +76,48 @@ class Map extends React.Component {
         const refresh = await Permissions.askAsync(Permissions.LOCATION);
         status = refresh.status;
       }
-      this.setState({ locationPermissions: status });
+      this.setLocationPermissions(status);
     }
     this.setState({ appState: nextAppState });
   };
 
   centerMapOnUserAsync = async () => {
-    let status = await this.getLocationPermissionsAsync();
+    await this.props.getLocationPermissionsAsync();
     let region = {
       latitude: 40.673868,
       longitude: -73.970089,
       latitudeDelta: 0.0922,
       longitudeDelta: 0.0421
     };
-    if (status === Constants.granted) {
-      let location = await Location.getCurrentPositionAsync({});
-      region = {
-        latitude: location.coords.latitude,
-        longitude: location.coords.longitude,
-        longitudeDelta: Constants.latLongDelta,
-        latitudeDelta: Constants.latLongDelta
-      };
+    if (this.props.locationPermissions === Constants.granted) {
+      region = await this.props.getUserLocationAsync();
     }
     this.props.setRegion(region);
   };
 
-  getLocationPermissionsAsync = async () => {
-    let { status } = await Permissions.askAsync(Permissions.LOCATION);
-    this.setState({ locationPermissions: status });
-    return status;
-  };
+  changeNearbyMapRegion (event) {
+    const { locationPermissions, setNearbyRegion } = this.props;
+    const { latitude, longitude } = event;
+    const region = {
+      latitude,
+      longitude,
+      latitudeDelta: Constants.latLongDelta,
+      longitudeDelta: Constants.latLongDelta
+    };
+
+    if (locationPermissions === Constants.denied) {
+      setNearbyRegion(region)
+    }
+  }
 
   setRegionAndSelectLandmark = (event, marker) => {
-    this.props.setRegion({
+    const region = {
       ...event.nativeEvent.coordinate,
       latitudeDelta: Constants.latLongDelta,
       longitudeDelta: Constants.latLongDelta
-    });
+    }
+    this.props.setRegion(region);
+    this.changeNearbyMapRegion(region);
```

Comment:
should be passing event into `changeNearbyMapRegion`?

Reply:
I don't believe so. We're using the data of `event` to create the `region` const so that function is getting the `event` info

Reply:
ohhh i see and then we pull lat/long off of the region and rebuild the region again. kinda confusing, but it makes sense to me now why it works.